code/modules/mob/living/living_movement.dm

```diff
@@ -26,38 +26,31 @@
 	return ..()
 
 /mob/living/proc/update_move_intent_slowdown()
-	var/mod = 0
-	if(m_intent == MOVE_INTENT_WALK)
-		mod = CONFIG_GET(number/movedelay/walk_delay)
-	else
-		mod = CONFIG_GET(number/movedelay/run_delay)
-	if(!isnum(mod))
-		mod = 1
-	add_movespeed_modifier(MOVESPEED_ID_MOB_WALK_RUN_CONFIG_SPEED, TRUE, 100, override = TRUE, multiplicative_slowdown = mod)
+	add_movespeed_modifier((m_intent == MOVE_INTENT_WALK)? /datum/movespeed_modifier/config_walk_run/walk : /datum/movespeed_modifier/config_walk_run/run)
 
 /mob/living/proc/update_turf_movespeed(turf/open/T)
-	if(isopenturf(T))
-		add_movespeed_modifier(MOVESPEED_ID_LIVING_TURF_SPEEDMOD, update=TRUE, priority=100, override=TRUE, multiplicative_slowdown=T.slowdown, movetypes=GROUND, blacklisted_movetypes=(FLYING|FLOATING))
+	if(istype(T))

```

Comment:
```suggestion
	if(isopenturf(T))
```
It's better if the check remains as is and `turf/open/T` gets changed to something like `atom/T` as else it's being misleading. The proc calling this uses `loc` on `Moved()`, and that can be any `atom`.